pkg/istio-agent/agent.go

```diff
@@ -320,33 +323,49 @@ func (a *Agent) Close() {
 // It may be different from the CA for the cert server - which is based on CA_ADDR
 // In addition it deals with the case the XDS server is on port 443, expected with a proper cert.
 // /etc/ssl/certs/ca-certificates.crt
-//
-// TODO: additional checks for existence. Fail early, instead of obscure envoy errors.
-func (a *Agent) FindRootCAForXDS() string {
+func (a *Agent) FindRootCAForXDS() (string, error) {
+	var rootCAPath string
+
 	if a.cfg.XDSRootCerts == security.SystemRootCerts {
-		return ""
+		// Special case input for root cert configuration to use system root certificates
+		return "", nil
 	} else if a.cfg.XDSRootCerts != "" {
-		return a.cfg.XDSRootCerts
-	} else if _, err := os.Stat("./etc/certs/root-cert.pem"); err == nil {
+		// Using specific platform certs or custom roots
+		rootCAPath = a.cfg.XDSRootCerts
+	} else if fileExists("./etc/certs/root-cert.pem") {
 		// Old style - mounted cert. This is used for XDS auth only,
 		// not connecting to CA_ADDR because this mode uses external
 		// agent (Secret refresh, etc)
-		return "./etc/certs/root-cert.pem"
+		return "./etc/certs/root-cert.pem", nil
 	} else if a.secOpts.PilotCertProvider == "kubernetes" {
 		// Using K8S - this is likely incorrect, may work by accident (https://github.com/istio/istio/issues/22161)
-		return k8sCAPath
+		rootCAPath = k8sCAPath
 	} else if a.secOpts.ProvCert != "" {
 		// This was never completely correct - PROV_CERT are only intended for auth with CA_ADDR,
 		// and should not be involved in determining the root CA.
-		return a.secOpts.ProvCert + "/root-cert.pem"
+		rootCAPath = a.secOpts.ProvCert + "/root-cert.pem"
```

Comment:
i think this file may be populated after this code is called for VMs

Reply:
Thanks! Skip checking existence for this case/possibility and add w/ comments.